### PR TITLE
EVG-2827: be able to degrade stats collection

### PR DESCRIPTION
**Pull request by tychoish**

### CODE REVIEW

File: operations/service.go [hunk 1]
```diff
@@ -53,3 +61,31 @@ func parseDB(c *cli.Context) *evergreen.DBSettings {
...
+		if err := flags.Get(); err != nil {
+			grip.Alert(message.WrapError(err, message.Fields{
+				"message":       "problem fetching service flags",
+				"operation":     "system stats",
+				"interval_secs": interval.Seconds(),
+			}))
+			return err
+		}
+
+		if flags.BackgroundStatsDisabled {
```
Comment (bsamek) [inline operations/service.go]: Did you mean to return early here?

File: operations/service_runner.go [hunk 2]
```diff
@@ -138,12 +139,35 @@ func startSystemCronJobs(ctx context.Context, env evergreen.Environment) {
...
+		if err := flags.Get(); err != nil {
+			grip.Alert(message.WrapError(err, message.Fields{
+				"message":       "problem fetching service flags",
+				"operation":     "background stats",
+				"interval_secs": backgroundStatsInterval.Seconds(),
+			}))
+			return err
+		}
+
+		if flags.BackgroundStatsDisabled {
```
Comment (bsamek) [inline operations/service_runner.go]: same question

File: operations/service_web.go [hunk 3]
```diff
@@ -130,23 +131,39 @@ func startWebService() cli.Command {
...
+		if err := flags.Get(); err != nil {
+			grip.Alert(message.WrapError(err, message.Fields{
+				"message":       "problem fetching service flags",
+				"operation":     "background stats",
+				"interval_secs": amboyStatsInterval.Seconds(),
+			}))
+			return err
+		}
+
+		if flags.BackgroundStatsDisabled {
```
Comment (bsamek) [inline operations/service_web.go]: same question

File: public/static/js/admin.js [hunk 4]
```diff
@@ -96,7 +96,8 @@ mciModule.controller('AdminSettingsController', ['$scope','$window', 'mciAdminRe
     github_pr_testing_disabled: "github_pr_testing",
     repotracker_push_event_disabled: "repotracker_push_event",
     cli_updates_disabled: "cli_updates",
-    github_status_api_disabled: "github_status_api"
+    github_status_api_disabled: "github_status_api",
+    background_stats_disabled: "bacground stats"
```
Comment (bsamek) [inline public/static/js/admin.js]: typo

File: service/templates/admin.html [hunk 5]
```diff
@@ -178,6 +178,12 @@ <h2 class="modal-title">[[modalTitle]]</h2>
                           <md-radio-button data-ng-value="false"></md-radio-button><md-radio-button data-ng-value="true"></md-radio-button>
                         </md-radio-group></td>
                       </tr>
+                      <tr>
+                        <td>Background Stats Collectors</td>
```
Comment (bsamek) [inline service/templates/admin.html]: Need to add github_status_api button too?

File: operations/service.go [hunk 6]
```diff
@@ -53,3 +61,31 @@ func parseDB(c *cli.Context) *evergreen.DBSettings {
...
 	}
 }
+
+////////////////////////////////////////////////////////////////////////
+//
+// Common Initialization Code
+
+func startSysInfoCollectors(ctx context.Context, env evergreen.Environment, interval time.Duration, opts amboy.QueueOperationConfig) {
+	amboy.IntervalQueueOperation(ctx, env.LocalQueue(), interval, time.Now(), opts, func(queue amboy.Queue) error {
+		flags := env.Settings().ServiceFlags
```
Comment (john-m-liu) [inline operations/service.go]: I would use evergreen.GetServiceFlags here

File: operations/service_runner.go [hunk 7]
```diff
@@ -138,12 +139,35 @@ func startSystemCronJobs(ctx context.Context, env evergreen.Environment) {
...
+	)
+
 	amboy.IntervalQueueOperation(ctx, env.RemoteQueue(), time.Minute, time.Now(), opts, units.PopulateActivationJobs())
 	amboy.IntervalQueueOperation(ctx, env.RemoteQueue(), 15*time.Minute, time.Now(), opts, units.PopulateCatchupJobs())
 	amboy.IntervalQueueOperation(ctx, env.RemoteQueue(), 90*time.Second, time.Now(), opts, units.PopulateRepotrackerPollingJobs())
 
 	// add jobs to a local queue every minute for stats collection and reporting.
-	amboy.IntervalQueueOperation(ctx, env.LocalQueue(), time.Minute, time.Now(), opts, func(queue amboy.Queue) error {
+	amboy.IntervalQueueOperation(ctx, env.LocalQueue(), backgroundStatsInterval, time.Now(), opts, func(queue amboy.Queue) error {
+		flags := env.Settings().ServiceFlags
```
Comment (john-m-liu) [inline operations/service_runner.go]: evergreen.GetServiceFlags()

File: operations/service_web.go [hunk 8]
```diff
@@ -130,23 +131,39 @@ func startWebService() cli.Command {
...
 		ContinueOnError: false,
 		LogErrors:       true,
 		DebugLogging:    false,
 	}
 
-	amboy.IntervalQueueOperation(ctx, env.LocalQueue(), time.Minute, time.Now(), opts, func(queue amboy.Queue) error {
+	const amboyStatsInterval = time.Minute
+
+	amboy.IntervalQueueOperation(ctx, env.LocalQueue(), amboyStatsInterval, time.Now(), opts, func(queue amboy.Queue) error {
+		flags := env.Settings().ServiceFlags
```
Comment (john-m-liu) [inline operations/service_web.go]: evergreen.GetServiceFlags()

File: service/templates/admin.html (hunk 5 above)
Comment (tychoish) [reply service/templates/admin.html]: what?

**Pull request merged**